kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/http/Interceptor.java

```diff
@@ -16,41 +16,23 @@
 
 package io.fabric8.kubernetes.client.http;
 
-import io.fabric8.kubernetes.client.Config;
-import io.fabric8.kubernetes.client.RequestConfig;
-
 import java.util.concurrent.CompletableFuture;
-import java.util.function.UnaryOperator;
 
 public interface Interceptor {
 
-  /**
-   * {@link Interceptor}s that rely upon the {@link Config}, in particular the {@link RequestConfig}, must implement
-   * this method to receive the modified configuration
-   *
-   * @param config
-   * @return
-   */
-  default Interceptor withConfig(Config config) {
-    return this;
-  }
+  public interface RequestTags {
+
+    <T> T getTag(Class<T> type);
```

Comment:
In the future we should document what a Request tag is and what can it be used for